packages/ui/src/store/index.ts

```diff
@@ -0,0 +1,44 @@
+import { FormEventHandler } from 'react';
+import create, { SetState, GetState } from 'zustand';
+
+import { signInBasic } from '@/apis/sign-in';
+
+export type PageState = 'idle' | 'loading' | 'error';
+
+export interface Store {
+  username: string;
+  password: string;
+  pageState: PageState;
+  setPassword: (password: string) => void;
+  setUsername: (username: string) => void;
+  signIn: FormEventHandler;
+}
+
+const useStore = create<Store>((set: SetState<Store>, get: GetState<Store>) => ({
```

Comment:
@wangsijie @gao-sun need some inputs on whether implement a single store for the entire SPA or one useStore for each individual page. Also in terms of define action slices separately.  Any best practice intension ?

Reply:
from what I'm practicing I would prefer create store for **shared** business logic/data. for this one, the store should be sth like 'UserStore' and the only property should be `username` (along with `setUsername()`). since `pageState` is per-page and `signIn` would be a hook that controls page state instead.